From 53e34e3bff26fcbb7cc14178fa9fc80e7a73d556 Mon Sep 17 00:00:00 2001
From: Baruch Siach <baruch@tkos.co.il>
Date: Tue, 11 Oct 2022 16:34:44 +0300
Subject: [PATCH] Makefile: disable stack protection

The Buildroot toolchain might enable stack protection by default. That
breaks linking because ATF does not provide the required __stack_chk
routines.

The mv-ddr-marvell Makefile provides no way to add custom CFLAGS. Patch
Makefile to disable stack protection.

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
Upstream status: not applicable; Buildroot specific
---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 3f0dd89a7381..feae75cc16e4 100644
--- a/Makefile
+++ b/Makefile
@@ -331,6 +331,7 @@ OBJ_DIR ?= $(MV_DDR_ROOT)
 CFLAGS = -DMV_DDR_ATF -DCONFIG_DDR4
 CFLAGS += -Wall -Werror -Os -ffreestanding -mlittle-endian -g -gdwarf-2 -nostdinc
 CFLAGS += -march=armv8-a -fpie
+CFLAGS += -fno-stack-protector
 
 # PLATFORM is set in ble/ble.mk
 ifneq ($(findstring a80x0,$(PLATFORM)),)
-- 
2.35.1

